lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -336,13 +336,21 @@ namespace Js
     bool
     FunctionBody::SkipAutoProfileForCoroutine() const
     {
-        return this->IsCoroutine() && CONFIG_ISENABLED(Js::JitES6GeneratorsFlag);
+#if defined(_ARM64_) || defined(_ARM_)
+        return false;
+#else
+        return this->IsCoroutine() && CONFIG_FLAG(JitES6Generators);
+#endif
     }
 
     bool
     FunctionBody::IsGeneratorAndJitIsDisabled() const
     {
-        return this->IsCoroutine() && !(CONFIG_ISENABLED(Js::JitES6GeneratorsFlag) && !this->GetHasTry() && !this->IsInDebugMode() && !this->IsAsync());
+#if defined(_ARM64_) || defined(_ARM_)
+        return this->IsCoroutine(); // disabled for ARM
+#else
+        return this->IsCoroutine() && !(CONFIG_FLAG(JitES6Generators) && !this->GetHasTry() && !this->IsInDebugMode() && !this->IsAsync());
```

Comment:
Curiosity from an outside contributor is there anything significant still keeping async funcitons (and async generators) from being jitted?

Reply:
Nothing significant really. async function should just work. Since I'm a summer intern and my task is to focus on enabling jitting for generators, disabling async generators initially helped me just focus on just one thing since there were some issues with the code in backend (which was fixed). Right now there are also some minor issues with other runtime optimizations for generators which also break async generators, so there will be a little more work to get it working.